embedded/devices/src/tca6416.cpp

```diff
@@ -0,0 +1,155 @@
+#include "tca6416.h"
+
+Tca6416::Tca6416(int bus, int addr) : I2c(bus, addr, O_RDWR) {}
+
+private:
+int Tca6416::clear_settings() {
+  /*Reset the configuration registers*/
+  if (this->write_data(TCA_CFG0_REG, 0x01) ||
+      this->write_data(TCA_CFG1_REG, 0x01))
+    return -1;
+
+  /*Reset the output registers*/
+  if (this->write_data(TCA_OUT0_REG, 0x01) ||
+      this->write_data(TCA_OUT1_REG, 0x01))
+    return -1;
+
+  /*Reset the input register*/
+  if (this->write_data(TCA_IN0_REG, 0xFF) ||
```

Comment:
The input register is read only according to the datasheet, it shouldn't need to be reset